src/nupic/algorithms/AnomalyLikelihood.cpp

```diff
@@ -0,0 +1,158 @@
+#include <iostream>
+#include <vector>
+#include <unordered_map>
+#include <string>
+
+#include <nupic/types/Types.hpp>
+
+using namespace std;
+
+struct DistributionParams {
+  DistributionParams(string name, Real32 mean, Real32 variance, Real32 stdev) :
+    name(name),mean(mean), variance(variance), stdev(stdev) {}
+  string name;
+  Real32 mean;
+  Real32 variance;
+  Real32 stdev;
+};
+
+struct DataPoint {
```

Comment:
I'd consider eliminating this struct and using passing all 3 required arguments to the respective methods. To keep the API same as in the python version.